mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/ChessFacade.kt

```diff
@@ -81,21 +83,27 @@ class ChessFacade(
    * @return The [Flow] of [List] of [Match]s for the [Profile]
    */
   fun matches(profile: Profile): Flow<List<Match>> {
-    val gamesAsWhite = getMatchesForPlayer(colorField = "whiteId", playerId = profile.uid)
-    val gamesAsBlack = getMatchesForPlayer(colorField = "blackId", playerId = profile.uid)
+    val gamesAsWhite = getMatchesForPlayer(colorField = "whiteId", playerId = profile.uid, profile)
+    val gamesAsBlack = getMatchesForPlayer(colorField = "blackId", playerId = profile.uid, profile)
 
     return combine(gamesAsWhite, gamesAsBlack) { (a, b) -> a.union(b).sortedBy { it.id } }
   }
 
-  private fun getMatchesForPlayer(colorField: String, playerId: String): Flow<List<Match>> {
-    return store.collection("games").whereEquals(colorField, playerId).asMatchListFlow().onStart {
-      emit(emptyList())
-    }
+  private fun getMatchesForPlayer(
+      colorField: String,
+      playerId: String,
+      user: Profile? = null
```

Comment:
What would you think of making the `user` argument mandatory and using `user.uid` to replace `playerId` ?

Reply:
Ah thx a lot cleaner